README.md

```diff
@@ -89,40 +89,73 @@ expectTypeOf({a: 1}).toEqualTypeOf({a: 2})
 expectTypeOf({a: 1, b: 1}).toEqualTypeOf<{a: number}>()
 ```
 
-To allow for extra properties, use `.toMatchTypeOf`. This is roughly equivalent to an `extends` constraint in a function type argument.:
+To allow for extra properties on an object type, use `.toMatchObjectType`. This is a strict check, but only on the subset of keys that are in the expected type:
 
 ```typescript
-expectTypeOf({a: 1, b: 1}).toMatchTypeOf<{a: number}>()
+expectTypeOf({a: 1, b: 1}).toMatchObjectType<{a: number}>()
 ```
 
-`.toEqualTypeOf` and `.toMatchTypeOf` both fail on missing properties:
+`.toMatchObjectType` can check partial matches on deeply nested objects:
+
+```typescript
+const user = {
+  email: 'a@b.com',
+  name: 'John Doe',
+  address: {street: '123 2nd St', city: 'New York', zip: '10001', state: 'NY', country: 'USA'},
+}
+
+expectTypeOf(user).toMatchObjectType<{name: string; address: {city: string}}>()
+```
+
+To check that a type extends another type, use `.toExtend`:
+
+```typescript
+expectTypeOf('some string').toExtend<string | boolean>()
+// @ts-expect-error
+expectTypeOf({a: 1}).toExtend<{b: number}>()
+```
+
+`.toExtend` can be used with object types, but `.toMatchObjectType` is usually a better choice when dealing with objects, since it's stricter:
+
+```typescript
+expectTypeOf({a: 1, b: 2}).toExtend<{a: number}>() // avoid this
+expectTypeOf({a: 1, b: 2}).toMatchObjectType<{a: number}>() // prefer this
+```
+
+`.toEqualTypeOf`, `.toMatchObjectType`, and `.toExtend` all fail on missing properties:
```

Comment:
As well as missing optional properties? (No time to try out. Sorry.)

Also I was wondering, why you don’t use `.not` in these examples? Because `// @ts-expect-error` makes these assertions pass also with older versions of this library.

Reply:
>As well as missing optional properties? (No time to try out. Sorry.)

`toEqualTypeOf` and `toMatchObjectType` fail on missing optional properties, but `toExtend` doesn't. Added some docs + tests to cover this explicitly though.

>why you don’t use .not in these examples

Just because these examples are essentially the docs and `.not` hasn't been introduced yet. I'd be open to reordering stuff to put `.not` higher up, you raise a good point - it's less prone to "wrong" errors.

Reply:
To be honest, it was my stupid mistake. I just copied this code to TS Playground and tried to mess up something. All worked really really well. Until.. I realised that this was because of `// @ts-expect-error`..